internal/agent/device/fileio/reader.go

```diff
@@ -7,46 +7,45 @@ import (
 )
 
 // Reader is a struct for reading files from the device
-type Reader struct {
+type reader struct {
 	// rootDir is the root directory for the device writer useful for testing
 	rootDir string
 }
 
 // New creates a new writer
-func NewReader() *Reader {
-	return &Reader{}
+func NewReader() *reader {
+	return &reader{}
 }
 
 // SetRootdir sets the root directory for the reader, useful for testing
-func (r *Reader) SetRootdir(path string) {
+func (r *reader) SetRootdir(path string) {
 	r.rootDir = path
 }
 
 // PathFor returns the full path for the provided file, useful for using functions
 // and libraries that don't work with the fileio.Reader
-func (r *Reader) PathFor(filePath string) string {
+func (r *reader) PathFor(filePath string) string {
 	return path.Join(r.rootDir, filePath)
 }
 
 // ReadFile reads the file at the provided path
-func (r *Reader) ReadFile(filePath string) ([]byte, error) {
+func (r *reader) ReadFile(filePath string) ([]byte, error) {
 	return os.ReadFile(r.PathFor(filePath))
 }
 
-// CheckPathExists checks if a path exists and will return an error if either
-// the file does not exist or if there is an error checking the path.
-func (r *Reader) CheckPathExists(filePath string) error {
+// FileExists checks if a path exists and returns a boolean indicating existence,
+// and an error only if there was a problem checking the path.
+func (r *reader) FileExists(filePath string) (bool, error) {
 	return checkPathExists(r.PathFor(filePath))
 }
 
-func checkPathExists(filePath string) error {
+func checkPathExists(filePath string) (bool, error) {
 	_, err := os.Stat(filePath)
-	if err != nil && os.IsNotExist(err) {
```

Comment:
i think this way it will be cleaner? WDYT?
`_, err := os.Stat(path)
    if err == nil { return true, nil }
    if os.IsNotExist(err) { return false, nil }
    return false, fmt.Errorf("error checking path: %w", err)`

Reply:
generally idiomatic to check !=nil first. but I don't love returning true at the end either will update.